### PR TITLE
Reduce test time for TensorRT EP CI

### PR DESCRIPTION
**Pull request by chilo-ms**
When updating to TRT 8.2, some tests require longer time to finish because new algorithms are added in TRT.
Following three modifications which are made specific to TRT EP CI can help reduce CI time by ~25 min on both Windows/Linux:
- Only run TRT ep for model tests and exclude all other eps. 
- Only run TRT ep with CUDA ep as fallback ep for unit tests.
- Reduction op test takes much longer time for TRT 8.2, so we test smaller range of inputs for TRT EP.

### CODE REVIEW

File: cmake/onnxruntime_unittests.cmake [hunk 1]
```diff
@@ -682,13 +682,19 @@ if (onnxruntime_BUILD_WEBASSEMBLY)
   endif()
 endif()
 
+set(test_all_args)
+if (onnxruntime_USE_TENSORRT)
+    list(APPEND test_all_args "--gtest_filter=-*cpu__*:*cuda__*" )
```
Comment (edgchen1) [inline cmake/onnxruntime_unittests.cmake:687]: comment on what we want to exclude here would be helpful.

Comment (chilo-ms) [reply cmake/onnxruntime_unittests.cmake:689]: I added comments here.

Comment (edgchen1) [reply cmake/onnxruntime_unittests.cmake:689]: I don't know where the "cpu__" and "cuda__" in the test names come from - I didn't see them in the code. Explaining that would be helpful.

Comment (chilo-ms) [reply cmake/onnxruntime_unittests.cmake:693]: Made the comment more clear.

    # The test names of model tests were using sequential number in the past.
    # This PR https://github.com/microsoft/onnxruntime/pull/10220 (Please see ExpandModelName function in model_tests.cc for more details) 
    # made test name contain the "ep" and "model path" information, so we can easily filter the tests using cuda ep or other ep with *cpu__* or *xxx__*.

File: onnxruntime/test/providers/provider_test_utils.cc [hunk 2]
```diff
@@ -1168,8 +1175,12 @@ void OpTester::Run(
         cur_provider = "not set";
       }
 
+#ifdef USE_TENSORRT
+      ORT_UNUSED_PARAMETER(has_run);
```
Comment (jywu-msft) [inline onnxruntime/test/providers/provider_test_utils.cc:1179]: add a comment here to explain this case here.
we are allowing tests to be run with *only* TensorRT EP, but TensorRT EP may not support all tests and may be in excluded providers list.

**Review by jywu-msft — DISMISSED** [review]

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**